keps/sig-storage/20191125-bucket-provisioning.md

```diff
@@ -188,21 +216,19 @@ metadata:
   finalizers:
   - cosi.io/finalizer [2]
 spec:
-  bucketName: [3]
-  generateBucketName: [4]
-  bucketClassName: [5]
-  secretName: [6]
-  protocol: [7]
-	accessMode: {"ro", "rw"} [8]
+  bucketPrefix: [3]
+  bucketClassName: [4]
+  secretName: [5]
+  protocol: [6]
+	accessMode: {"ro", "rw"} [7]
 status:
-  bucketContentName: [9]
+  bucketContentName: [8]
   phase:
   conditions: 
 ```
 1. `labels`: COSI controller adds the label to its managed resources to easy CLI GET ops.  Value is the driver name returned by GetDriverInfo() rpc*.
 1. `finalizers`: COSI controller adds the finalizer to defer Bucket deletion until backend deletion ops succeed.
-1. `bucketName`: Desired name of the bucket to be created**.
-1. `generateBucketName`: Desired prefix to a randomly generated bucket name. Mutually exclusive with `bucketName`**.
+1. `bucketPrefix`: prefix prepended to a randomly generated bucket name. API validation _may_ ensure that `Bucket` namespace + `bucketPrefix` + generated name is less than the max length of a metadata.name.
```

Comment:
> `Bucket` namespace + `bucketPrefix` + generated name

Using the namespace is probably overkill if we're generating a random suffix anyway.

Reply:
That was a mistake on my part. I confused the actual bucket name with our proposed name of the `BucketContent` object. Will remove.